src/interfaces/IAppliance.js

```diff
@@ -0,0 +1,95 @@
+import {
+	InterfaceInstantiationError,
+	NotImplementedError,
+} from '../errors'
+
+class IAppliance {
+	/**
+	 * Constructor for a new appliance
+	 * @param  {Object} overrideSettings Values to override default appliance settings.
+	 */
+	// eslint-disable-next-line no-unused-vars
+	constructor(overrideSettings = {}) {
+		if (this.constructor === IAppliance) {
+			throw new InterfaceInstantiationError(this.constructor.name)
+		}
+	}
+
+	/**
+	 * Getter for the list of data types accepted by the appliance.
+	 *
+	 * @return {String[]} The list of data types accepted by the appliance.
+	 */
+	getInputTypes = () => {
+		throw new NotImplementedError('getInputTypes')
+	}
+
+	/**
+	 * Getter for the list of data types produced by the appliance.
+	 *
+	 * @return {String[]} The list of data types produced by the appliance.
+	 */
+	getOutputTypes = () => {
+		throw new NotImplementedError('getOutputTypes')
+	}
+
+	/**
+	 * Asserts that a given payload is actually a payload.
+   *
+	 * @param  {Payload} payload The payload that we want to validate.
+	 * @return {Boolean}         The result of the validation check.
+	 */
+	// eslint-disable-next-line no-unused-vars
+	isValidPayload = async (payload) => {
+		throw new NotImplementedError('isValidPayload')
+	}
+
+	/**
+	 * Invokes the appliance on any unprocessed data in the appliance buffer.
+	 *
+	 * @return {Boolean} Whether or not the appliance had enough data to work with.
+	 */
+	invoke = async () => {
+		throw new NotImplementedError('invoke')
+	}
+
+	/**
+	 * Called to pass data into the appliance.
+	 * If the payload is valid, it is added to the buffer and the appliance is invoked.
```

Comment:
Quibble: Either make this line part of the line above it, or put another newline between them. Personally I like the format of "Very succinct opening description, followed by sensibly-defined expanded descriptions below." In this case, I feel like the second line is succinct enough it could be part of the first line, but your call.

I'm being insanely quibbly about formatting/standards/etc. at this point since it's easier to quibble _now_ when the first violations are occurring than _later_ when we have to clean a bunch up.